solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -1599,7 +1599,13 @@ public ScoreMode scoreMode() {
       ScoreMode scoreModeUsed = buildAndRunCollectorChain(qr, query, collector, cmd, pf.postFilter).scoreMode();
 
       totalHits = topCollector.getTotalHits();
-      TopDocs topDocs = topCollector.topDocs(0, len);
+      TopDocs topDocs = null;
+      try {
+        topDocs = topCollector.topDocs(0, len);
+      } catch (ReRankCollector.UnReRankedTopDocs urrTopDocs) {
+        topDocs = urrTopDocs.topDocs;
+        qr.setPartialResults(true);
+      }
```

Comment:
`buildAndRunCollectorChain` already handles the timeout exception but afterwards `topCollector.topDocs` can also encounter the timeout exception. If un-reranked documents are available then we provide them here and indicate to the caller that the response is partial. The `timeAllowed` query parameter [1] and the `partialResults` response flag go together i.e. the client would already know to check about it. Also this would apply to both reranking via `ltr` and reranking via the `rerank` [2] parser code paths.

[1] https://solr.apache.org/guide/8_8/query-re-ranking.html#rerank-query-parser
[2] https://solr.apache.org/guide/8_8/common-query-parameters.html#timeallowed-parameter